### PR TITLE
Added a new Tic Tac Toe game made using C

### PR DESCRIPTION
**Pull request by vivekboss99**
I thought it would be good to include a folder containing simple games we can make using C ,so i have made few games using C like Tic-Tac-Toe, Sudoku puzzle generator, etc. I think it brings out the creativity in any programmer while making games and also it's like the starting steps to any beginner programmer to who wants to develop video games.  

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/530"></a>

### CODE REVIEW

**Comment by vivekboss99** [comment]
Updated Code Formatting and removed .DS_Store files

**Comment by kvedala** [comment]
Looks good 👍

**Comment by kvedala** [comment]
Good algorithm and I’d like to see it included. If you can update the documentation, you can create a pull request to : [this repo](https://github.com/kvedala/C.git) and I can review it

**Comment by kvedala** [comment]
This implementation gets into an indefinite loop calling the function `place()` and running in to segmentation fault.

File: games/tic-tac-toe.c [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+static void placey(int);  // used in Double Player mode by the 2nd player to
+                          // place the position of O
+int checkwin();  // checks everytime when a player or computer places X or O
+
+/** Tic-Tac-Toe table, so basically we are using variable 'a' as the table and
+ * updating it regularly
+ */
+static char game_table[9];
+
+int main()
```
Comment (kvedala) [inline games/tic-tac-toe.c:31]:
```suggestion
/**
 * Main program function.
 * @returns 0 on clean exit. 
 * @note No checks are included for program execution failures!
 */
int main()
```

File: games/tic-tac-toe.c [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        scanf("%d", &l);
+
+    } while (l == 1);
+
+    return 0;
+}
+
+/**
+ * @brief Implementation of game vs computer
+ */
```
Comment (kvedala) [inline games/tic-tac-toe.c:80]:
```suggestion
 * @returns None
 */
```

File: games/tic-tac-toe.c [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+        {
+            printf("\nDRAW ");
+            break;
+        }
+    }
+}
+
+/**
+ * @brief Implementation of game vs another player.
+ *
```
Comment (kvedala) [inline games/tic-tac-toe.c:149]:
```suggestion
 * @returns None
```

File: games/tic-tac-toe.c [hunk 4]
```diff
@@ -298,7 +313,11 @@ void placey(int e1)
         placey(n1);
     }
 }
-
+/**
+ * @brief Implementation of win conditon checker for 'X' or 'O' whenever the table is updated
+ *
+ * @returns -1: if 'X' won , -2: if 'O' won , 0: if there is no win condition for 'X' or 'O'
```
Comment (kvedala) [inline games/tic-tac-toe.c:319]:
```suggestion
 * @returns -1: if 'X' won 
 * @returns -2: if 'O' won
 * @returns 0: if there is no win condition for 'X' or 'O' 
```
a minor fix.

**Pull request merged**